addons/hw_drivers/drivers/SerialScaleDriver.py

```diff
@@ -0,0 +1,284 @@
+
+ScaleProtocol = namedtuple('ScaleProtocol', SerialProtocol._fields + ('zeroCommand', 'tareCommand', 'clearCommand'))
+
+# 8217 Mettler-Toledo (Weight-only) Protocol, as described in the scale's Service Manual.
+#    e.g. here: https://www.manualslib.com/manual/861274/Mettler-Toledo-Viva.html?page=51#manual
+# Our recommended scale, the Mettler-Toledo "Ariva-S", supports this protocol on
+# both the USB and RS232 ports, it can be configured in the setup menu as protocol option 3.
+# We use the default serial protocol settings, the scale's settings can be configured in the
+# scale's menu anyway.
+Toledo8217Protocol = ScaleProtocol(
+    name='Toledo 8217',
+    baudrate=9600,
+    bytesize=serial.SEVENBITS,
+    stopbits=serial.STOPBITS_ONE,
+    parity=serial.PARITY_EVEN,
+    timeout=1,
+    writeTimeout=1,
+    measureRegexp=b"\x02\\s*([0-9.]+)N?\\r",
+    statusRegexp=b"\x02\\s*(\\?.)\\r",
+    commandDelay=0.2,
+    measureDelay=0.5,
+    newMeasureDelay=0.2,
+    commandTerminator=b'',
+    measureCommand=b'W',
+    zeroCommand=b'Z',
+    tareCommand=b'T',
+    clearCommand=b'C',
+    emptyAnswerValid=False,
+    autoResetWeight=False,
+    disable=False
+)
+
+# The ADAM scales have their own RS232 protocol, usually documented in the scale's manual
+#   e.g at https://www.adamequipment.com/media/docs/Print%20Publications/Manuals/PDF/AZEXTRA/AZEXTRA-UM.pdf
+#          https://www.manualslib.com/manual/879782/Adam-Equipment-Cbd-4.html?page=32#manual
+# Only the baudrate and label format seem to be configurable in the AZExtra series.
+ADAMEquipmentProtocol = ScaleProtocol(
+    name='Adam Equipment',
+    baudrate=4800,
+    bytesize=serial.EIGHTBITS,
+    stopbits=serial.STOPBITS_ONE,
+    parity=serial.PARITY_NONE,
+    timeout=0.2,
+    writeTimeout=0.2,
+    measureRegexp=b"\s*([0-9.]+)kg",  # LABEL format 3 + KG in the scale settings, but Label 1/2 should work
+    statusRegexp=None,
+    commandTerminator=b"\r\n",
+    commandDelay=0.2,
+    measureDelay=0.5,
+    # AZExtra beeps every time you ask for a weight that was previously returned!
+    # Adding an extra delay gives the operator a chance to remove the products
+    # before the scale starts beeping. Could not find a way to disable the beeps.
+    newMeasureDelay=5,
+    measureCommand=b'P',
+    zeroCommand=b'Z',
+    tareCommand=b'T',
+    clearCommand=None,  # No clear command -> Tare again
+    emptyAnswerValid=True,  # AZExtra does not answer unless a new non-zero weight has been detected
+    autoResetWeight=True,  # AZExtra will not return 0 after removing products
+    disable=True
+)
+
+
+# Ensures compatibility with older versions of Odoo
+class OldScaleDriver(http.Controller):
+    @http.route('/hw_proxy/scale_read', type='json', auth='none', cors='*')
+    def scale_read(self):
+        if ACTIVE_SCALE:
+            ACTIVE_SCALE._scale_read_old_route()
+            return {'weight': ACTIVE_SCALE.data['value']}
+        return None
```

Comment:
do we want the backward compatibility?
also, this kind of things should be clearly explained in the commit message

Reply:
Yes we really need it. I explained it in the commit message.